internal/impl/opensearch/output.go

```diff
@@ -0,0 +1,454 @@
+	}
+	if e.typeStr, err = mgr.BloblEnvironment().NewField(conf.Type); err != nil {
+		return nil, fmt.Errorf("failed to parse type field expression: %v", err)
+	}
+
+	for _, u := range conf.URLs {
+		for _, splitURL := range strings.Split(u, ",") {
+			if len(splitURL) > 0 {
+				e.urls = append(e.urls, splitURL)
+			}
+		}
+	}
+
+	if tout := conf.Timeout; len(tout) > 0 {
+		var err error
+		if e.timeout, err = time.ParseDuration(tout); err != nil {
+			return nil, fmt.Errorf("failed to parse timeout string: %v", err)
+		}
+	}
+
+	if e.backoffCtor, err = conf.Config.GetCtor(); err != nil {
+		return nil, err
+	}
+
+	if conf.TLS.Enabled {
+		var err error
+		if e.tlsConf, err = conf.TLS.Get(); err != nil {
+			return nil, err
+		}
+	}
+	return &e, nil
+}
+
+//------------------------------------------------------------------------------
+
+// Connect attempts to establish a connection to a OpenSearch broker.
+func (e *OpenSearch) Connect(ctx context.Context) error {
+	if e.client != nil {
+		return nil
+	}
+	boff := e.backoffCtor()
+	opts := os.Config{
+		Addresses:     e.urls,
+		RetryOnStatus: []int{502, 503, 504, 429},
+		RetryBackoff: func(attempt int) time.Duration {
+			if attempt == 1 {
+				boff.Reset()
+			}
+			return boff.NextBackOff()
+		},
+		MaxRetries: int(e.conf.MaxRetries),
+	}
+	if e.conf.Auth.Enabled {
+		opts.Username = e.conf.Auth.Username
+		opts.Password = e.conf.Auth.Password
+	}
+
+	opts.Transport = &http.Transport{
+		ResponseHeaderTimeout: e.timeout,
+	}
+	if e.conf.TLS.Enabled {
+		opts.Transport = &http.Transport{
+			TLSClientConfig:       e.tlsConf,
+			ResponseHeaderTimeout: e.timeout,
+		}
+
+	}
+
+	awsOpts, err := AWSOptFn(opts.Transport, e.conf)
+	if err != nil {
+		return err
+	}
+	if awsOpts != nil {
+		opts.Transport = awsOpts
+
+	}
+
+	if e.conf.GzipCompression {
+		opts.CompressRequestBody = true
+	}
+
+	client, err := os.NewClient(opts)
+	if err != nil {
+		return err
+	}
+
+	e.client = client
+	e.log.Infof("Sending messages to OpenSearch index at urls: %s\n", e.urls)
+	return nil
+}
+
+type pendingBulkIndex struct {
+	Action   string
+	Index    string
+	Pipeline string
+	Routing  string
+	Type     string
+	Doc      interface{}
+	ID       string
+}
+
+// WriteBatch will attempt to write a message to OpenSearch, wait for
+// acknowledgement, and returns an error if applicable.
+func (e *OpenSearch) WriteBatch(ctx context.Context, msg message.Batch) error {
+	return e.Write(msg)
+}
+
+// Write will attempt to write a message to OpenSearch, wait for
+// acknowledgement, and returns an error if applicable.
+func (e *OpenSearch) Write(msg message.Batch) error {
+	if e.client == nil {
+		return component.ErrNotConnected
+	}
+
+	requests := make([]*pendingBulkIndex, msg.Len())
+	if err := msg.Iter(func(i int, part *message.Part) error {
+		jObj, ierr := part.AsStructured()
+		if ierr != nil {
+			e.log.Errorf("Failed to marshal message into JSON document: %v\n", ierr)
```

Comment:
This seems a bit redundant, but I see it's [the same](https://github.com/benthosdev/benthos/blob/ef8fb69ad85a3b7311bacaa78565b8a6b5127ead/internal/impl/elasticsearch/output.go#L275) in the ES output. Might be a leftover that we should clean up (cc @Jeffail)

Reply:
This might be required for opensearch go client as the update requires interface{} object.